graphdatascience/aura_api.py

```diff
@@ -91,7 +92,16 @@ def __init__(self, client_id: str, client_secret: str, tenant_id: Optional[str]
         self._logger = logging.getLogger()
         self._tenant_id = tenant_id if tenant_id else self._get_tenant_id()
 
-    def create_instance(self, name: str) -> InstanceCreateDetails:
+    @classmethod
+    def extract_id(cls, uri: str) -> str:
+        host = urlparse(uri).hostname
+
+        if not host:
+            raise RuntimeError(f"Could not parse the uri {uri}.")
+
+        return host.split(".")[0].split("-")[0]
```

Comment:
there is a risk that this parsing fails, right? array index out of bounds etc. or does `urlparse` guarantee that the output supports this splitting? we should add unit tests and if we can write one that fails here then we should catch and throw

Reply:
urlparse returns a string and as we always takes the very first after the split, at worst you endup with an invalid string `id`. But I dont think we want to check constraints on the id.

added some test though :)